docs: add clearer documentation for multiple cluster services (#2243)
karmada-io/karmada: docs/multi-cluster-service.md

### Karmada has been installed

> We can install Karmada by referring to [Quick Start](https://github.com/karmada-io/karmada#quick-start), or directly run `hack/local-up-karmada.sh` script which is also used to run our e2e cases.

Reviewer: Is there a requirement for the case of `e2e`?

Author: I understand the description here to mean that you can use 'hack/ local-up-karmda.sh' to deploy Karmada, and this script was also used in e2e testing

Reviewer: Sorry, I didn't make it clear, I was actually referring to the case format.

Author: OK, do you mean keep E2E?

We need to install ServiceExport and ServiceImport in the member clusters.

> After ServiceExport and ServiceImport have been installed on the Karmada Control Plane, we can create `ClusterPropagationPolicy` to propagate those two CRDs to the member clusters.

Reviewer: We have bolded these keywords, why should we change them?

Author: I noticed that in README, 'Karmada Control Plane' was used, so I changed it to the same as README

Reviewer: Thanks for your explanation, but I think that the focus may be different in different contexts, and I think it can be kept inconsistent.

Author: Okay, can I add the bold?

### Step 2: Export service to the `member2` cluster

- Create a `ServiceExport` object on **Karmada Control Plane**, and then create a `PropagationPolicy` to propagate the `ServiceExport` object to the `member1` cluster.

  ```yaml
  apiVersion: multicluster.x-k8s.io/v1alpha1
  kind: ServiceExport
  metadata:
    name: serve
  ---
  apiVersion: policy.karmada.io/v1alpha1
  kind: PropagationPolicy
  metadata:
    name: serve-export-policy
  spec:
    resourceSelectors:
      - apiVersion: multicluster.x-k8s.io/v1alpha1
        kind: ServiceExport
        name: serve
    placement:
      clusterAffinity:
        clusterNames:
          - member1
  ```

> - Create a `ServiceImport` object on **Karmada Control Plane**, and then create a `PropagationPlicy` to propagate the `ServiceImport` object to the `member2` cluster.

Reviewer 2: ```suggestion
- Create a `ServiceImport` object on **Karmada Control Plane**, and then create a `PropagationPolicy` to propagate the `ServiceImport` object to the `member2` cluster.
```

  ```yaml
  apiVersion: multicluster.x-k8s.io/v1alpha1
  kind: ServiceImport
  metadata:
    name: serve
  spec:
    type: ClusterSetIP
    ports:
    - port: 80
      protocol: TCP
  ---
  apiVersion: policy.karmada.io/v1alpha1
  kind: PropagationPolicy
  metadata:
    name: serve-import-policy
  spec:
    resourceSelectors:
      - apiVersion: multicluster.x-k8s.io/v1alpha1
        kind: ServiceImport
        name: serve
    placement:
      clusterAffinity:
        clusterNames:
          - member2
  ```

### Step 3: Access the service from `member2` cluster

After the above steps, we can find the **derived service** which has the prefix `derived-` on the `member2` cluster. Then, we can access the **derived service** to access the service on the `member1` cluster.
```shell

> # get service on the member2 cluster, we can find the service with the name 'derived-serve'

Reviewer 2: ```suggestion
# get the services in cluster member2, and we can find the service with the name 'derived-serve'
```

$ kubectl --kubeconfig ~/.kube/members.config --context member2 get svc
NAME            TYPE        CLUSTER-IP    EXTERNAL-IP   PORT(S)   AGE
derived-serve   ClusterIP   10.13.205.2   <none>        80/TCP    81s
kubernetes      ClusterIP   10.13.0.1     <none>        443/TCP   15m
```

Start a pod `request` on the `member2` cluster to access the ClusterIP of **derived service**:

```shell
$ kubectl --kubeconfig ~/.kube/members.config --context member2 run -i --rm --restart=Never --image=jeremyot/request:0a40de8 request -- --duration={duration-time} --address={ClusterIP of derived service}
```

Eg, if we continue to access service for 3s, ClusterIP is `10.13.205.2`:

```shell

> # access the service of derived-serve, the pod in member1 cluster returns a response

Reviewer 2: ```suggestion
# access the service of derived-serve, and the pod in member1 cluster returns a response
```
